[Revived] Multiblock Tooltips GTLanguageManager -> MC lang system port

ChromaPIE opened the pull request:
Revives https://github.com/GTNewHorizons/GT5-Unofficial/pull/4206, based on the latest commit, neat commit history (compared to the former one in chaos).

TO-DOs (might need help):
- **Backward compatibility** so players benefitting from localization projects won't have to suffer from their game turning into a whole Duolingo course overnight. A surgery on `MultiblockTooltipBuilder` is needed for this one because the implementation now gets the job done but might be a bit too radical. It might be what we eventually need but yeah, we can't expect loc contributors to work like dynamos unpaid.
  By Backward I mean those yet to be ported will still base the way their tooltips are handled on the Manager, but for the ported ones (I've ported the **Purification Plant** and the **IsaMill** as a showcase, more will come later), so yeah, localization is still needed. But y'all might want this PR to contain the entire port for each and every MBs before it gets merged, this is also nice.
- **Auto separator length & paged tooltips compat**
  **Related methods**: `getTooltipWidth` and `applySeparatorLine` of `ItemMachines`
  As of now the length of separators in each tooltip = the longest tooltip line. This leads to a problem that if the tooltip goes out of the screen and thus gets paged, and the longest text line is not on the currect page, all separators can be drastically longer. (Yes. don't look away. Eye of Harmony. I'm talking about you) So we might need a different approach and make the separator = the longest line on the CURRENT page, and able to dynamically adjust.
- **Machine Type Standardization**: Machine type should only include recipe types a machine can handle, like Dangote has both Distillery and Distillation Tower, but quite a bunch of MBs include their acronyms in the machine type for some reason. I don't really want to just wipe the acronyms off the tooltips but they should be displayed somewhere else, just not as a separate machine type entry. Because machine type, at least in this refactor, should not be for acronyms nor aliases. It should reflect what groups of recipes an MB is able to process.
  Anyway, Acceptable cases include:
  - the acronym is exactly short for the type, like `Quantum Force Transformer, QFT`, as `,` is not really a hard separator unlike `|` or so;
  - It simply uses the short form for actual separate recipe types. Like for Dangote, if it writes `Distillery | DT`, that's ok.

**CAUTION:** some tooltips use keys from GregTech.lang generated by the Manager, basically for the casings. I'm not porting them to MC lang system because it'd cause a good headache and that I'm not bringing this PR too far from just MB tooltips, but future Manager deprecation related to those casings should also come visit and edit the keys in the tooltips, like
```diff
- .addCasingInfoMin("gtplusplus.blockcasings.5.4.name", 45)
+ .addCasingInfoMin("gtplusplus.whatever_youd_call_it.name", 45)
```
**EDIT:** It's also recommended to directly fetch the names from the code for a better consistency, like:
```
.addCasingInfoMin(
    ItemRefer.ProtomatterActivationCoil.get(1)
        .getDisplayName(),
    126)
```
or
```
.addCasingInfoExactly(Casings.FilterMachineCasing.getLocalizedName(), 21)
```

**Improvements, notes, questions & further additions etc.**
- One better thing about it is that it minimizes lines that has to wait for translation for each machine as we now directly use the already translated casing names, recipe category names etc., and quite some simply multiblockified singleblock machines leave zero entries to be translated. The whole thing gets drastically unified and simplified.
- `"<hint>"` as info for structure parts to display the part info as `(Hatch/Bus Type): Hint block with dot 1, 3, 5` (numbers grabbed from the `int...` parameter) was there, but never used, maybe bc it's newly added. This PR would encourage people to use it rather than writing their own info, which makes it worse in standardization among tooltips.
- `"<casing>"` as info for `Any Casing` (`GT5U.MBTT.AnyCasing`) in `addStructurePart`
- Removed all `Author: `s from author tags in GTValues. It'll be prepended at the final stage. Removed check and substring due to this
- Do I still have to put the lang lines in each multiblock's respective lang file or just shove them all together into the gregtech one (what I'm doing now)?

Check the old PR mentioned above for more info.

ChromaPIE commented:
> Why couldn't we just copy over the old translations to the new keys, then update ParaTranz to use the new keys?

1. If you really checked things you can see how it's differentiated in handling info lines. We've been using multiple `addInfo`s for each tooltip section separated by the separation line but now in this PR they get merged into one single line in the lang file, with line breaks marked by `\n`. Simply changing the key isn't possible anymore at this point. In fact I don't really think ParaTranz even supports batch swapping keys for existing translated texts but still. And it still takes a whole lotta manual work, might be even more tedious.
2. The separation line. Yes, all separation lines got their own lang key. This is terribly stupid.
3. It's not sheerly about just copy pasting translations. It's a refactor to the tooltip builder and some other stuff.

ChromaPIE committed:
cleanups + deprecate manual separation line length

ChromaPIE committed:
remove Author: from GTValues

ChromaPIE committed:
Windmill & EOH

where it starts + where it ends

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
don't know what it does but, cleanup

StructureWrapperTooltipBuilder

ChromaPIE committed:
BHC done

and its util hatch

ChromaPIE committed:
MB input hatch/bus desc tooltips

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
upstream change

ChromaPIE committed:
more hatches

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
BBF, cleanroom

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
TFFT, SO Fuel Cell MK1&2 improved

ChromaPIE committed:
dyson swarm

shout out to the great practice

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
new helper method

ChromaPIE committed:
mVF

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

Nikolay-Sitnikov commented on src/main/java/gregtech/api/enums/Materials.java:
```diff
@@ -1919,6 +2002,10 @@ public String getLocalizedNameForItem(String aFormat) {
+    public String getLocalizedNameForItem() {
+        return getLocalizedNameForItem("%material");
+    }
+
```
Isn't this equivalent to `return aFormat`?

ChromaPIE replied:
it returns `this.mLocalizedName` actually. So yeah maybe I should just delete it and use like `.getLocalizedName()`

I did bc one class or two did `getLocalizedNameForItem("%material")`

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
I.O.F. & dangote

ChromaPIE commented:
@Nikolay-Sitnikov I don't really plan to go into the `%material` part further, at least not in this PR so yup let's just leave pro russian stuff for russian pros to deal with later.

ChromaPIE committed:
fix

ChromaPIE committed:
use recipe group name first for machine type

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
HILE & LATEX & small improvements

ChromaPIE committed:
HILE & LATEX & small improvements

ChromaPIE committed:
Merge branch 'MBTT-newke' of https://github.com/ChromaPIE/GT5-Unofficial into MBTT-newke

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
EIC, LATEX fix

ChromaPIE committed:
GTValues EnumChatFormatting cleanup

use string friendly statics

ChromaPIE committed:
goofy ahh author related cleanups Part.1

ChromaPIE committed:
DEHP & cleanups

ChromaPIE committed:
DEHP

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Merge branch 'MBTT-newke' of https://github.com/ChromaPIE/GT5-Unofficial into MBTT-newke

ChromaPIE committed:
more bart author stuff cleanups

ChromaPIE committed:
improve TTBuilder

no need for manual indents anymore

ChromaPIE committed:
fix `GTUtility.translate` related error text

ChromaPIE committed:
HTGR + EIG

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
DEFC+EEC+MApiary

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
sync upstream TT update

ChromaPIE committed:
multi autoclave

ChromaPIE committed:
nano forge

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
ind electrolyzer, ind freezer, ind arc furnace

and stuff and stuff

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Industrial Alloyer, Hammer, Washer

ChromaPIE committed:
i mean..

we seriously should NOT put acronyms into machine type. bc they are not ones.

ChromaPIE committed:
ind wiremill

ChromaPIE committed:
sifter. dehp fixie.

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
merge adjust

ChromaPIE committed:
ECE, entropic

ChromaPIE committed:
ECE, entropic

ChromaPIE committed:
Merge branch 'MBTT-newke' of https://github.com/ChromaPIE/GT5-Unofficial into MBTT-newke

ChromaPIE committed:
3 fusion reactors

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
upstream

ChromaPIE committed:
charcoal pit

ChromaPIE committed:
Time for a revision

ChromaPIE committed:
revision 2

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
upstream

ChromaPIE committed:
dtower, coke oven

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
mb boilers

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
assline, spinmatron, nano forge fixie

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
air filters

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Merge branch 'MBTT-newke' of https://github.com/ChromaPIE/GT5-Unofficial into MBTT-newke

ChromaPIE committed:
Separator refactrator

No need for `.addSeparator()` calls, use `<SEPARATOR>` in lang.

ChromaPIE committed:
The Great Semanticization - The First Step

ChromaPIE committed:
structure sep line too

ChromaPIE committed:
keep it going

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
isamill dedigit

ChromaPIE committed:
ULTRA cleanup

ChromaPIE committed:
FIX

ChromaPIE committed:
port

ChromaPIE committed:
fix indent

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
lil enum rename

ChromaPIE committed:
this and that

ChromaPIE committed:
fixie

ChromaPIE committed:
fix color code handling

was chopping off the last 2 chars

ChromaPIE committed:
JSON loc parsing, rename appendParams

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
this and that

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
this and that

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
semanticize

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
spotless

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
fusion reactors

ChromaPIE committed:
Merge branch 'MBTT-newke' of https://github.com/ChromaPIE/GT5-Unofficial into MBTT-newke

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
upstream

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
GET casing names pt.1

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
upstream

ChromaPIE committed:
Refinery, Decay, Fixes & More Names From Code

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
upstream

ChromaPIE committed:
unused imports

ChromaPIE committed:
Merge branch 'MBTT-newke' of https://github.com/ChromaPIE/GT5-Unofficial into MBTT-newke

ChromaPIE committed:
Update TooltipMarkupProcessor.java

ChromaPIE committed:
keep refactoring

ChromaPIE committed:
fix <S_SEPARATOR> indent

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
upstream

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Solar Factory & Fixes

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
tpm & wormhole gen

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
spotless

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
fix

ChromaPIE committed:
DTPF

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
spotless

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
spotless n stuff

ChromaPIE committed:
fix build

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
add debugTooltips

ChromaPIE committed:
I-Electrolyzer, IWF, Thermal Boiler

ChromaPIE committed:
code style

ChromaPIE committed:
use enums from casing.Casings

clean and tidy

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
fix

ChromaPIE committed:
fix, multi canner

ChromaPIE committed:
only append space for :

dont need one for full width ：

ChromaPIE committed:
working on NAC

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Revert "only append space for :"

This reverts commit 568d520d0d51a8cb3176e32352e6701899405b5b.

ChromaPIE committed:
not the space

ChromaPIE committed:
format StructureBy

ChromaPIE committed:
pushing

ChromaPIE committed:
Merge branch 'MBTT-newke' of https://github.com/ChromaPIE/GT5-Unofficial into MBTT-newke

ChromaPIE committed:
pushing

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
cpi

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
clarifier

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
antimatter forge

ChromaPIE committed:
refactor displayed hint info stuff

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
adjustments

ChromaPIE committed:
antimatter forge done

ChromaPIE committed:
Merge branch 'MBTT-newke' of https://github.com/ChromaPIE/GT5-Unofficial into MBTT-newke

ChromaPIE committed:
name from oreprefix & material enum only

ChromaPIE committed:
use GTOreDictUnificator#getLocalizedName

Eldrinn-Elantey committed:
refactor(drills): migrate hardcoded strings to localization keys

Replace direct StatCollector calls with GTUtility.translate() across
MTEOilDrillBase and MTEOreDrillingPlantBase.

Move hardcoded tooltip and info strings into en_US.lang:
- Oil drill machine type, description, and structure info
- Ore drill machine type and description
- Shared keys for min energy hatch tier, base cycle time, controller
  position, and structure part placements

Fix double colon in GT5U.MBTT.Controller lang entry — the colon was
duplicated because PartInfo format string already appends ": " between
the part name and its positional info.

Use BlockFrameBox.getLocalizedName() to resolve frame material name
in the oil drill tooltip instead of concatenating mName directly.

ChromaPIE commented on src/main/java/gregtech/common/tileentities/machines/multi/MTEOilDrillBase.java:
```diff
@@ -117,28 +118,32 @@ public void loadNBTData(NBTTagCompound aNBT) {
     protected MultiblockTooltipBuilder createTooltip(String tierSuffix) {
         String casings = getCasingBlockItem().get(0)
             .getDisplayName();
+        String frameBoxes = ((BlockFrameBox) GregTechAPI.sBlockFrames).getLocalizedName(getFrameMaterial());
```
In 886513eec76c0ece8c7ac3d3a9fdcb14b8e832b5 I added a new method in `src\main\java\gregtech\api\util\GTOreDictUnificator.java` so we can use `GTOreDictUnificator.getLocalizedName(OrePrefixes.frameGt, getFrameMaterial())`

ChromaPIE committed:
MTEOilDrillBase polish

yeah maybe we can unify `Base cycle time` stuff and reuse them. we'll see.

ChromaPIE committed:
ore drill

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
groupie

ChromaPIE committed:
SLAM, SC Turbine, YOTTank

ChromaPIE committed:
semantizing n polishing

ChromaPIE committed:
LHE

ChromaPIE committed:
use gt.mbtt.structure. to adapt to upstream

ChromaPIE committed:
LCE

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
UCFE

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
IOF redone

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Neutron Activator

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Nq Generator

ChromaPIE committed:
PrAss, various tidy-ups

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
pushing

ChromaPIE committed:
more cleanups

ChromaPIE committed:
deprecate addManyInfo

ChromaPIE committed:
impressor, bbb, multi centrifuge

ChromaPIE committed:
LST, other adjustments

ChromaPIE committed:
MegaCR

ChromaPIE committed:
Casings instead of String for anyCasingText

ChromaPIE committed:
That's How We Tidy Things Up

ChromaPIE committed:
front_bottom_CENTER

ChromaPIE committed:
fix build

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
manual trafo, biovat, stuff n stuff

ChromaPIE committed:
THTR, other cleanups

ChromaPIE committed:
ass family

ChromaPIE committed:
GTUtility.translate in MTEMultiBlockBaseGui & de-hardcoding

ChromaPIE committed:
dehardcoding "/each"

ChromaPIE committed:
cal finished

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Update GTAuthors.java

ChromaPIE committed:
dealing with all these bullsh

ChromaPIE committed:
spotless

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
upstream

ChromaPIE committed:
shrinking further

ChromaPIE committed:
MEBF

ChromaPIE committed:
MOC

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
spotless

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
upstream

ChromaPIE committed:
digester, dissolution tank

ChromaPIE committed:
tieredGlassText

ChromaPIE committed:
another great example of how this PR tidies things up

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
upstream

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
FoG stuff

ChromaPIE committed:
research station, tesla tower

ChromaPIE committed:
tt stuff wip

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
pushing

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
upstream

ChromaPIE committed:
Merge upstream/master into MBTT-newke

ChromaPIE committed:
Sync migrated MTE tooltips with upstream

ChromaPIE committed:
Merge latest upstream/master into MBTT-newke

ChromaPIE committed:
Fix delayed localization in multiblock tooltip APIs

ChromaPIE committed:
upd

ChromaPIE committed:
Fix structure tooltip placeholder resolution

ChromaPIE committed:
Merge branch 'master' into MBTT-newke

ChromaPIE committed:
upd

ChromaPIE committed:
Fix subchannel tooltip localization

ChromaPIE committed:
Show standalone structure authors

ChromaPIE committed:
Simplify delayed tooltip localization plumbing

ChromaPIE committed:
Fix tooltip cleanup compilation and formatting

ChromaPIE committed:
upd

ChromaPIE committed:
Localize coolant tower tooltip

ChromaPIE committed:
spotless

ChromaPIE committed:
Localize TFFT tooltip

ChromaPIE committed:
Localize CoAL tooltip
